### ISSUE TITLE
Cant deploy this contract on Test Network - out of gas

### ISSUE THREAD

**Issue by creotiv** [makerdao/maker-otc#169]
Trying to deploy this contract to Genache test network, but getting out of gas every time. Even with higher limits it still not deploying. It seems like it have forever loop or something.

For deploy im using Truffle. Would be appreciate for help.

**Comment by gbalabasquer** [comment]
did you try to deploy to some of the testnets? like kovan? to discard there is a specific issue with ganache.

**Comment by creotiv** [comment]
So the only way i was able to push contracts to the test net is to remove
contract size limitation.

Kindly yours,
Andrey Nikishaev

Areas         ML/DS/CV/Soft Dev/BizDev/Growth Hacking/Customer Rel/IT
LinkedIn      http://ua.linkedin.com/in/creotiv
GitHub        http://github.com/creotiv
Slideshare  https://www.slideshare.net/anikishaev/
Skype         creotiv.in.ua
Mobile        +380632410666

**Comment by gbalabasquer** [comment]
did you compile the contract using `--optimize`?

**Comment by creotiv** [comment]
No, will test it and reply

Kindly yours,
Andrey Nikishaev

Areas         ML/DS/CV/Soft Dev/BizDev/Growth Hacking/Customer Rel/IT
LinkedIn      http://ua.linkedin.com/in/creotiv
GitHub        http://github.com/creotiv
Slideshare  https://www.slideshare.net/anikishaev/
Skype         creotiv.in.ua
Mobile        +380632410666

**Comment by creotiv** [comment]
optimize not helped, still exceeds block gas limit

Kindly yours,
Andrey Nikishaev

Areas         ML/DS/CV/Soft Dev/BizDev/Growth Hacking/Customer Rel/IT
LinkedIn      http://ua.linkedin.com/in/creotiv
GitHub        http://github.com/creotiv
Slideshare  https://www.slideshare.net/anikishaev/
Skype         creotiv.in.ua
Mobile        +380632410666

**Comment by gbalabasquer** [comment]
Just made a test deploy to kovan. It succeed, 3.5 M gas used.
You are probably doing something wrong with optimize param. I'm using dapptools, but you could copy the code in Remix and try it there. Remember to paste all the dependencies.

**Comment by creotiv** [comment]
Im using truffle compile with solc params:

solc: {
optimizer: {
enabled: true,
runs: 200
}
},

Kindly yours,
Andrey Nikishaev

Areas         ML/DS/CV/Soft Dev/BizDev/Growth Hacking/Customer Rel/IT
LinkedIn      http://ua.linkedin.com/in/creotiv
GitHub        http://github.com/creotiv
Slideshare  https://www.slideshare.net/anikishaev/
Skype         creotiv.in.ua
Mobile        +380632410666

**Comment by gbalabasquer** [comment]
Yeah, that seems correct. Not sure why it is not working with Truffle, but if exceeds the block limit is because the optimisation of the tool is not being applied for some reason.
